tests/js/client/shell/shell-med-queue-blocked-cluster.js

```diff
@@ -0,0 +1,139 @@
+/* global fail, assertEqual, assertTrue, assertFalse, arango */
+
+// //////////////////////////////////////////////////////////////////////////////
+// / @brief timeouts during query setup
+// /
+// / DISCLAIMER
+// /
+// / Copyright 2023 ArangoDB GmbH, Cologne, Germany
+// /
+// / Licensed under the Apache License, Version 2.0 (the "License")
+// / you may not use this file except in compliance with the License.
+// / You may obtain a copy of the License at
+// /
+// /     http://www.apache.org/licenses/LICENSE-2.0
+// /
+// / Unless required by applicable law or agreed to in writing, software
+// / distributed under the License is distributed on an "AS IS" BASIS,
+// / WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// / See the License for the specific language governing permissions and
+// / limitations under the License.
+// /
+// / @author Max Neunhoeffer
+// //////////////////////////////////////////////////////////////////////////////
+
+let jsunity = require('jsunity');
+let arangodb = require('@arangodb');
+let db = arangodb.db;
+let { getEndpointsByType,
+      debugCanUseFailAt,
+      debugClearFailAt,
+      debugSetFailAt,
+    } = require('@arangodb/test-helper');
+      
+function medQueueBlockedSuite() {
+  'use strict';
+  const cn = 'UnitTestsMedQueueBlocked';
+
+  return {
+
+    setUp: function () {
+      // We are using the direct connection to set the failure point here,
+      // since for this particular failure point (block scheduler medium
+      // queue) a reconnect is not possible. Therefore we cannot use
+      // debugFailAt.
+      arango.DELETE_RAW("/_admin/debug/failat");
+      db._drop(cn);
+    },
+
+    tearDown: function () {
+      // We are using the direct connection to set the failure point here,
+      // since for this particular failure point (block scheduler medium
+      // queue) a reconnect is not possible. Therefore we cannot use
+      // debugFailAt.
+      arango.DELETE_RAW("/_admin/debug/failat");
+      require("internal").wait(3);  // give the system time to finish the 
+                                    // creation of the collection, in case
+                                    // the test TestCreateColl fails.
+      db._drop(cn);
+    },
+    
+    testRunAql: function() {
+      let coll = db._create(cn, {numberOfShards:3});
+      let l = [];
+      for (let i = 0; i < 100; ++i) {
+        l.push({"Hello":i});
+      }
+      coll.insert(l);
+
+      // We now want to overload the medium prio scheduler queue, to this
+      // end, we post some perverse queries asynchronously. This should
+      // overwhelm any server without the fix. Note that we post the query
+      // with x-arango-frontend: true, such that we can start enough queries
+      // concurrently to fill the medium queue on the coordinator.
+      const f = require("@arangodb/aql/functions")
+      f.register("USER::FUNC", function() {
```

Comment:
We should unregister this AQL function either at the end of this test or in the `tearDown` function.
Otherwise this AQL function may leak into other tests and cause trouble there.

Reply:
Correct, will do.

Reply:
Done.